bazelisk.go

```diff
@@ -391,20 +471,20 @@ func determineURL(fork string, version string, isCommit bool, filename string) s
 		kind = "rc" + versionComponents[1]
 	}
 
-	if fork == bazelUpstream {
+	if bazeliskConfig.useUpstream {
 		return fmt.Sprintf("https://releases.bazel.build/%s/%s/%s", version, kind, filename)
 	}
 
-	return fmt.Sprintf("https://github.com/%s/bazel/releases/download/%s/%s", fork, version, filename)
+	return fmt.Sprintf("https://%s/%s/bazel/releases/download/%s/%s", bazeliskConfig.githubUrl, bazeliskConfig.owner, version, filename)
```

Comment:
General comment:  I think the changes are focused too much on github in the loop, here we can see githubUrl (incomplete for the purposes of a download url) and github users are central to determining the download url.  A more basic assumption would just be a template url that just needs the version interpolated in, letting the user decide if its https, or github - or perhaps even file:// path'd.  Borrowing from http_archive and friends in bazel land, it would be better if this was determineURLs too - so mirrors and filepaths could both be offered (perhaps ordering file-paths based earlier)



Also this is forcing https where as validation looks for http